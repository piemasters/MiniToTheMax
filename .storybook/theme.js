import { create } from "@storybook/theming/create";
import logo from "../content/assets/images/icon.png";


const theme = create({
    base: "light",

    colorPrimary: "#ff5470",
    colorSecondary: "#00dcb5",

    // UI
    appBg: "white",
    appContentBg: "white",
    appBorderColor: "grey",
    appBorderRadius: 10,

    // Typography
    fontBase: '"Open Sans", sans-serif',
    fontCode: "monospace",

    // Text colors
    textColor: "black",
    textInverseColor: "rgba(255,255,255,0.9)",

    // Toolbar default and active colors
    barTextColor: "white",
    barSelectedColor: "#00dcb5",
    barBg: "#00153d",

    // Form colors
    inputBg: "white",
    inputBorder: "white",
    inputTextColor: "black",
    inputBorderRadius: 4,

    brandTitle: "MiniToTheMax Storybook",
    brandUrl: "https://minitothemax.app",
    brandImage: logo
});

export default theme;