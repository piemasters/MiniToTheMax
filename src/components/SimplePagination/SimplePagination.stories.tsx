import type { Meta, StoryObj } from '@storybook/react-vite';

import { SimplePagination } from './SimplePagination';

const meta: Meta<typeof SimplePagination> = {
  component: SimplePagination,
  argTypes: {
    previous: {
      name: 'prevPage',
      description: 'Link to the previous page',
      defaultValue: { slug: 'previous', title: 'Previous Page' },
      control: { type: 'object' },
    },
    next: {
      name: 'nextPage',
      description: 'Link to the next page',
      defaultValue: { slug: 'next', title: 'Next Page' },
      control: { type: 'object' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof SimplePagination>;

export const Primary: Story = {
  args: {
    previous: {
      slug: 'previous',
      title: 'Previous Page',
    },
    next: {
      slug: 'next',
      title: 'Next Page',
    },
  },
};
