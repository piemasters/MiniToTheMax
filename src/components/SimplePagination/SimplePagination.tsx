import React, { FC } from 'react';

import { PageLink } from '../PageLink';

type SimplePaginationNode = {
  slug: string;
  title: string;
};

export interface SimplePaginationProps {
  previous?: SimplePaginationNode;
  next?: SimplePaginationNode;
}

export const SimplePagination: FC<SimplePaginationProps> = ({
  previous,
  next,
}) => {
  return (
    <div
      data-testid="simple-pagination"
      className="flex flex-wrap justify-between list-none p-0 no-underline"
    >
      <div className="flex-1 py-8 pr-8">
        {next && (
          <PageLink to={next.slug} type={'cover'} direction={'left'}>
            <div className="flex text-right">
              <div className="pr-2">← </div>
              <div className="flex-1">{next.title}</div>
            </div>
          </PageLink>
        )}
      </div>
      <div className="flex-1 py-8 pl-8">
        {previous && (
          <PageLink to={previous.slug} type={'cover'} direction={'right'}>
            <div className="flex">
              <div className="flex-1">{previous.title}</div>
              <div className="pl-2">→</div>
            </div>
          </PageLink>
        )}
      </div>
    </div>
  );
};

export default SimplePagination;
