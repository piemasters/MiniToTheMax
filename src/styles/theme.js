export default {
  colors: {
    text: '#000',
    muted: '#999',
  },
};
