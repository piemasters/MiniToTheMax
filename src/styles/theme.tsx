export interface Theme {
  colors: {
    text: string;
    textSecondary: string;
    textLight: string;
    hyperlink: string;
    hyperlinkActive: string;
    muted: string;
    lightgrey: string;
    lightgreyHover: string;
    primary: string;
    backgroundTransparent: string;
  };
}

const appTheme: Theme = {
  colors: {
    text: '#000',
    textSecondary: '#666',
    textLight: 'rgba(255, 255, 255, 0.9)',
    hyperlink: '#4f61ee',
    hyperlinkActive: '#0231e9',
    muted: '#999',
    lightgrey: '#ececec',
    lightgreyHover: 'e4e4e4',
    primary: '#eb1d23',
    backgroundTransparent: 'rgba(0, 0, 0, 0.7)',
  },
};

export default appTheme;
