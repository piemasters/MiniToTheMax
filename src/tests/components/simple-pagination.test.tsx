import React from 'react';
import SimplePagination from '../../components/simple-pagination';
import { render, getByTestId } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../styles/theme';

describe('SimplePagination', () => {
  test('renders correctly', () => {
    const previous = {
      slug: 'previous',
      title: 'The Previous Page',
    };

    const next = {
      slug: 'next',
      title: 'The Next Page',
    };

    const { container } = render(
      <ThemeProvider theme={theme}>
        <SimplePagination previous={previous} next={next} />
      </ThemeProvider>
    );

    expect(getByTestId(container, 'simple-pagination')).toHaveTextContent(
      previous.title
    );
    expect(getByTestId(container, 'simple-pagination')).toHaveTextContent(
      next.title
    );
    expect(container).toMatchSnapshot();
  });
});
