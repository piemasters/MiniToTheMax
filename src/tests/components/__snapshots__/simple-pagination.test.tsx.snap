// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePagination renders correctly 1`] = `
<div>
  <div
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="simple-pagination"
  >
    <div
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <a
        bg="#eb1d23"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        data-testid="page-link-cover"
        direction="right"
        duration="1"
        href="/previous"
      >
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            ← 
          </div>
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            The Previous Page
          </div>
        </div>
      </a>
    </div>
    <div
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <a
        bg="#eb1d23"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        data-testid="page-link-cover"
        direction="left"
        duration="1"
        href="/next"
      >
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            The Next Page
          </div>
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            →
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
